include/IECore/StringAlgo.inl

```diff
@@ -213,7 +194,18 @@ inline bool matchMultiple( const std::string &s, const MatchPattern &patterns )
 
 inline bool matchMultiple( const char *s, const char *patterns )
 {
-	return Detail::matchInternal( s, patterns, /* multiple = */ true );
+	do
+	{
+		if( Detail::matchInternal( s, patterns, /* spaceTerminates = */ true ) )
+		{
+			return true;
+		}
+		// Advance to next pattern.
+		patterns += strcspn( patterns, " " ); // Increment to first space
+		patterns += strspn( patterns, " " ); // Increment to next non-space
```

Comment:
It seems like there's enough test coverage that this must be working, but I don't understand exactly how termination works.  Based on the docs for strcspn - it says that unlike strspn it includes the null terminator, which makes it sound like this could advance past the final null terminator?

Reply:
Which docs are you reading? Neither my man pages nor cppreference.com say anything to imply that :

https://en.cppreference.com/w/c/string/byte/strcspn
https://en.cppreference.com/w/c/string/byte/strspn

Perhaps you're looking at http://www.cplusplus.com/reference/cstring/strcspn/?

> The search includes the terminating null-characters. Therefore, the function will return the length of str1 if none of the characters of str2 are found in str1.

I think this is just awkward wording based on the awkward description preceding it. The second sentence seems to clarify things though - it won't bounce you off the end, it'll just increment you to point exactly at the null terminator, which is what we want.

Reply:
@danieldresser-ie and I discussed this separately yesterday and concluded that the documentation is just a bit misleading, so I think we're good to go on this one.